pallets/stateful-storage/src/tests/itemized_operations_tests.rs

```diff
@@ -0,0 +1,184 @@
+}
+
+#[test]
+fn applying_delete_action_with_existing_index_should_delete_item() {
+	// arrange
+	let payloads = vec![
+		generate_payload_bytes::<ItemizedBlobSize>(Some(2)),
+		generate_payload_bytes::<ItemizedBlobSize>(Some(4)),
+	];
+	let page = create_itemized_page_from::<Test>(None, payloads.as_slice());
+	let expecting_page = create_itemized_page_from::<Test>(None, &payloads[1..]);
+	let actions = vec![ItemAction::Delete { index: 0 }];
+
+	// act
+	let result = ItemizedOperations::<Test>::apply_item_actions(&page, &actions);
+
+	// assert
+	assert_ok!(&result);
+	let updated = result.unwrap();
+	assert_eq!(expecting_page.data, updated.data);
+}
+
+#[test]
+fn applying_add_action_should_add_item_to_the_end_of_the_page() {
+	// arrange
+	let payload1 =
+		vec![generate_payload_bytes::<<Test as Config>::MaxItemizedBlobSizeBytes>(Some(2))];
+	let page = create_itemized_page_from::<Test>(None, payload1.as_slice());
+	let payload2 =
+		vec![generate_payload_bytes::<<Test as Config>::MaxItemizedBlobSizeBytes>(Some(4))];
+	let expecting_page = create_itemized_page_from::<Test>(
+		None,
+		&vec![payload1[0].clone(), payload2[0].clone()][..],
+	);
+	let actions = vec![ItemAction::Add { data: payload2[0].clone().into() }];
+
+	// act
+	let result = ItemizedOperations::<Test>::apply_item_actions(&page, &actions[..]);
+
+	// assert
+	assert_ok!(&result);
+	let updated = result.unwrap();
+	assert_eq!(expecting_page.data, updated.data);
+}
+
+#[test]
+fn applying_delete_action_with_non_existing_index_should_fail() {
+	// arrange
+	let payloads = vec![
+		generate_payload_bytes::<ItemizedBlobSize>(Some(2)),
+		generate_payload_bytes::<ItemizedBlobSize>(Some(4)),
+	];
+	let page = create_itemized_page_from::<Test>(None, payloads.as_slice());
+	let actions = vec![ItemAction::Delete { index: 2 }];
+
+	// act
+	let result = ItemizedOperations::<Test>::apply_item_actions(&page, &actions[..]);
+
+	// assert
+	assert_eq!(result.is_err(), true);
+}
+
+#[test]
+fn applying_add_action_with_full_page_should_fail() {
+	// arrange
+	let payload: Vec<u8> = vec![1u8; ItemizedBlobSize::get() as usize];
+	let size_to_fill = (<Test as Config>::MaxItemizedPageSizeBytes::get() as usize) -
+		sp_std::mem::size_of::<PageNonce>();
+	let mut item_buffer = Vec::<u8>::with_capacity(size_to_fill);
+	while (size_to_fill as i32).saturating_sub(item_buffer.len().try_into().unwrap()) >
+		ItemizedBlobSize::get().try_into().unwrap()
+	{
+		add_itemized_payload_to_buffer::<Test>(&mut item_buffer, &payload.as_slice());
+	}
+	let page: ItemizedPage<Test> =
+		ItemizedPage::<Test> { nonce: 0, data: item_buffer.try_into().unwrap() };
+	let actions = vec![ItemAction::Add { data: payload.try_into().unwrap() }];
+
+	// act
+	let result = ItemizedOperations::<Test>::apply_item_actions(&page, &actions[..]);
+
+	// assert
+	assert_eq!(result, Err(PageError::PageSizeOverflow));
+}
+
+#[test]
+fn apply_delete_item_on_append_only_fails() {
```

Comment:
I think this should be in apply_item_actions file and the test name might also need to change to use the convention

Reply:
There are several.  You previously wrote: "the ones starting with parsing_item and applying_delete or applying_add are ItemizedOperations". Let me know if you want to keep them in this file or change to the apply_items_actions file.

Reply:
If it was testing ` ItemizedOperations::<Test>::apply_item_actions` I would have kept it here

Reply:
The distinction is what is triggered in the` // act` section of the test

Reply:
Therefore I will move `apply_item_actions_existing_page_with_stale_hash_should_fail()` and `apply_item_actions_with_signature_having_page_with_stale_hash_should_fail()` to `itemized_operations_tests.rs`.